### PR TITLE
make the description editable

### PR DESCRIPTION
**Pull request by binayak20**
- Making a different component to update the description
- Making a general function to update all the project fields
- If you suggest me to replace the updateTitle with new common function I can do this

### CODE REVIEW

File: components/editable-description.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  const [originalDescription, setOriginalDescription] = useState(
+    project.description || "Untitled",
+  );
+  const [editing, setEditing] = useState(false);
+
+  async function clickUpdateProject() {
+    try {
+      const updates = { description: description };
+      await updateProject(project.id, updates);
+      setOriginalDescription(description);
```
Comment (patrickchin) [inline components/editable-description.tsx:34]: here I think we should use the returned value of updateProject

Comment (binayak20) [reply components/editable-description.tsx]: fix all the issues

File: components/editable-description.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+          <div className="mt-6 flex items-center justify-end gap-x-3">
+            <Button
+              variant="default"
+              className="p-2"
+              onClick={clickUpdateProject}
+            >
+              Save
+            </Button>
+            <Button variant="secondary" className="p-2" onClick={cancelEditing}>
+              Cancel
```
Comment (patrickchin) [inline components/editable-description.tsx:86]: tiny request, flip the order of the save and cancel buttons

File: components/editable-description.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+  function cancelEditing() {
+    setDescription(originalDescription); // Revert to original title
+    setEditing(false);
+  }
+
+  return (
+    <>
+      <CardHeader>
+        <div className="flex items-center gap-1">
```
Comment (patrickchin) [inline components/editable-description.tsx:50]: justify-between and the edit button variant secondary
I think it looks a bit better that way :)

File: components/editable-description.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+      </CardHeader>
+      <CardContent>
+        <CardDescription className="whitespace-pre-line">
+          {!editing ? (
+            description
+          ) : (
+            <Textarea
+              defaultValue={description}
+              placeholder="Tell us a little bit about your project"
+              className="resize-y h-svh"
```
Comment (patrickchin) [inline components/editable-description.tsx:71]: ideally we'd want the height to match the content exactly, but it seems like shadcn/ui hasn't merged that functionality yet

https://github.com/shadcn-ui/ui/pull/1681

if you so desire you could copy paste this in, but definitely not necessary


I think svh is too high now :/ maybe h-72 is enough

**Review by patrickchin — CHANGES_REQUESTED** [review]
LGTM, just a few tiny more tweaks, and a rebase onto main, and it's good to merge

File: components/editable-description.tsx (hunk 4 above)
Comment (binayak20) [reply components/editable-description.tsx:71]: I have fixed this issue in different approach. If you like you can keep this.

File: components/editable-description.tsx [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+  useEffect(() => {
+    const textarea = textareaRef.current;
+    if (textarea && editing) {
+      textarea.style.height = "auto";
+      textarea.style.height = `${textarea.scrollHeight}px`;
+      textarea.style.overflowY = "hidden";
+    }
+  }, [editing, description]);
+
+  const handleChangeDescription = (event: any) => {
```
Comment (patrickchin) [inline components/editable-description.tsx:39]: prefer `ChangeEvent<HTMLTextAreaElement>` over any

**Review by patrickchin — APPROVED** [review]
Looks great and works really well!!

**Pull request merged**